src/components/TagInput.js

```diff
@@ -0,0 +1,126 @@
+import React, { Component } from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import uid from 'uid';
+import * as color from './res/colors'
+
+const TagInputWrapper = styled.div`
+  position: relative;
+  margin-bottom: 8px;
+`
+
+const TagInputElement = styled.input`
+    font-size: 1em;
+    outline: none;
+    min-width: 180px;
+    border: 1px solid ${color.active};
+    margin: 0 2px;
+    padding: 2px 8px;
+    border-radius: 3px;
+    background: inherit; 
+  `
+
+const SuggestionsWrapper = styled.div`
+  position: absolute;
+  z-index: 1;
+  min-width: 184px;
+  background: ${color.white};
+  border: 1px solid ${color.lineDark};
+  margin: 0 2px;
+  padding: 2px 8px;
+`
+
+const Suggestion = styled.div`
+  padding: 2px 0;
+`
+
+
+export default class TagInput extends Component {
+
+  inputElement = React.createRef()
+
+  state = {
+    searchString: '',
+    suggestions: [],
+  }
+
+  getSuggestions = value => {
+    const { suggestableTags, hasTopic } = this.props
+    const inputValue = value.trim().toLowerCase()
+    const inputLength = inputValue.length
+    return inputLength === 0 ? [] : suggestableTags
+      .filter(tag => tag.topic !== hasTopic)
+      .filter(item => item.name.toLowerCase().slice(0, inputLength) === inputValue
```

Comment:
Extrahiere diesen zweiten Filter in eine Arrow-Funktion mit Namen direkt als `const` in dieser Funktion. Damit man weiß, was sie tut.

Reply:
Meinst Du etwa so? : 
```
  getSuggestions = value => {
    const { suggestableTags, hasTopic } = this.props
    const inputValue = value.trim().toLowerCase()
    const inputLength = inputValue.length
    const matchInputValue = tag => tag.name.toLowerCase().slice(0, inputLength) === inputValue
    return inputLength === 0 ? [] : suggestableTags
      .filter(tag => tag.topic !== hasTopic)
      .filter(tag => matchInputValue(tag))
  }
```